### PR TITLE
App SSO NodeJS

### PR DESCRIPTION
**Pull request by Tanmay-MSFT**
Description
This sample app illustrates Bot Tab ME SSO Scenario.

Language Used: javascript/node js

Proposed features
Teams Tab, Bot, ME - search, action, linkunfurl SSO with Node JS

Testing
You can interact with this bot by sending it a message. The bot will respond by requesting you to login to AAD, then making a call to the Graph API on your behalf and returning the results.
- Install App

image

- Type *anything* on the compose box and send
- The bot will perform `Single Sign-On` and Profile card will be displayed along with the option prompt to view the `token`

image

> **NOTE:** 
If the user is using the application for the first time and user consent is required for additional permissions, the following dialog box appears to continue with the consent experience
image
image

>If the bot couldn't perform `SSO` then it will fallback to normal Authentication method and show a `Sign In` card like below
image

- Open `Messaging Extension`(Search), it will show profile details

image
image
image

- Open `Messaging Extension`(Action), it will show profile details

image

__*or*__

image
image

- Open `Messaging Extension`(linkunfurl), The link will unfurl and show profile details

**Paste** https://profile.botframework.com on the compose box

image
image

> NOTE: If `SSO` couldn't be performed then it will fallback to normal Authentication method and you will get a default `Sign In` action

Consent the *ME Search* by clicking the `Sign In` link like below 

image

Consent the *ME Action* by clicking the `Setup` button like below 

image

- Open `SSO Tab`, Continue and then Accept and it'll show the profile details

image
image
image

### CODE REVIEW

File: samples/app-sso/nodejs/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Teams Tab, Bot, ME - search, action, linkunfurl SSO with Node JS
+
+__Tab SSO__
+This sample shows how to implement Azure AD single sign-on support for tabs. It will
+
+- Obtain an access token for the logged-in user using SSO
```
Comment (Trinetra-MSFT) [inline samples/app-sso/nodejs/README.md:6]: Please add screenshot to setup connection string in App

File: samples/app-sso/nodejs/teamsAppManifest/manifest.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    "websiteUrl": "https://example.azurewebsites.net",
+    "privacyUrl": "https://example.azurewebsites.net/privacy",
+    "termsOfUseUrl": "https://example.azurewebsites.net/termsofuse"
+  },
+  "icons": {
+    "color": "icon-color.png",
+    "outline": "icon-outline.png"
+  },
+  "name": {
+    "short": "Bot Tab ME SSO",
```
Comment (Trinetra-MSFT) [inline samples/app-sso/nodejs/teamsAppManifest/manifest.json:22]: Please update name to App SSO Node, mention in readMe as well

**Review by Trinetra-MSFT — CHANGES_REQUESTED** [review]
Please update root ReadMe file

File: samples/app-sso/nodejs/README.md (hunk 1 above)
Comment (Tanmay-MSFT) [reply samples/app-sso/nodejs/README.md:6]: Entire setup is mentioned under 

## To try this sample

- Setup for Bot SSO
Refer to [Bot SSO Setup document](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-conversation-sso-quickstart/BotSSOSetup.md).

Please have a look and let me know if anything needs to be added

File: samples/app-sso/nodejs/teamsAppManifest/manifest.json (hunk 2 above)
Comment (Tanmay-MSFT) [reply samples/app-sso/nodejs/teamsAppManifest/manifest.json]: Updated the Readme and manifest.
Please have a look and let me know if anything needs to be added

File: samples/app-sso/nodejs/README.md (hunk 1 above)
Comment (Trinetra-MSFT) [reply samples/app-sso/nodejs/README.md:8]: Please mark review comment resolve once you commit changes

**Comment by Trinetra-MSFT** [comment]
@Tanmay-MSFT : I can't find this sample link in Root ReadMe file could you please add this in scenario specific samples

**Review by Trinetra-MSFT — APPROVED** [review]

File: samples/app-sso/nodejs/README.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    ngrok http -host-header=localhost 4000
+    ```
+
+
+- Update the `.env` configuration for the bot to use the `MicrosoftAppId` (Microsoft App Id), `MicrosoftAppPassword` (App Password) and `connectionName` (OAuth Connection Name) from the Bot Framework registration. 
+> NOTE: the App Password is referred to as the `client secret` in the azure portal and you can always create a new client secret anytime.
+
+
+- Install modules & Run the Server `Bot, ME`
+Open a terminal and navigate to project root directory
```
Comment (Wajeed-msft) [inline samples/app-sso/nodejs/README.md:81]: Is it possible to combine these under same project for simplicity?

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/app-sso/nodejs/README.md (hunk 3 above)
Comment (Tanmay-MSFT) [reply samples/app-sso/nodejs/README.md]: PR has been updated with Updated Readme and Screen shots, please have a look

Comment (Tanmay-MSFT) [reply samples/app-sso/nodejs/README.md]: Updated PORT to 3978 and readme file

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**